st2common/st2common/util/crypto.py

```diff
@@ -13,40 +13,291 @@
+with SHA1 HMAC signature) as keyczar methods, but they use and rely on primitives and methods from
+the cryptography library.
+
+This was done to make the keyczar -> cryptography migration fully backward compatible.
+
+Eventually, we should  move to Fernet (https://cryptography.io/en/latest/fernet/) recipe for
+symmetric encryption / decryption, because it offers more robustness and safer defaults (SHA256
+instead of SHA1, etc.).
+"""
+
 from __future__ import absolute_import
+
+import os
+import json
 import binascii
+import base64
+
+from hashlib import sha1
+
+import six
+
+from cryptography.hazmat.primitives.ciphers import Cipher
+from cryptography.hazmat.primitives.ciphers import algorithms
+from cryptography.hazmat.primitives.ciphers import modes
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import hmac
+from cryptography.hazmat.backends import default_backend
 
 __all__ = [
+    'KEYCZAR_HEADER_SIZE',
+    'KEYCZAR_AES_BLOCK_SIZE',
+    'KEYCZAR_HLEN',
+
     'read_crypto_key',
+
     'symmetric_encrypt',
-    'symmetric_decrypt'
+    'symmetric_decrypt',
+
+    'cryptography_symmetric_encrypt',
+    'cryptography_symmetric_decrypt',
+
+    # NOTE: Keyczar functions are here for testing reasons - they are only used by tests
+    'keyczar_symmetric_encrypt',
+    'keyczar_symmetric_decrypt',
+
+    'AESKey'
 ]
 
+# Keyczar related constants
+KEYCZAR_HEADER_SIZE = 5
+KEYCZAR_AES_BLOCK_SIZE = 16
+KEYCZAR_HLEN = sha1().digest_size
+
 
-def read_crypto_key(key_path, key_type=None):
+class AESKey(object):
+    """
+    Class representing AES key object.
     """
-    Return the crypto key given a path to key file and the key type.
 
-    :param key_path: Absolute path to file containing crypto key.
-    :type key_path: ``str``
+    aes_key_string = None
+    hmac_key_string = None
+    hmac_key_size = None
+    mode = None
+    size = None
 
-    :param key_type: Type of crypto key.
-    :type key_type: :class:`keyczar.keys.KeyType`
+    def __init__(self, aes_key_string, hmac_key_string, hmac_key_size, mode='CBC', size=128):
+        if mode not in 'CBC':
+            raise ValueError('Unsupported mode: %s' % (mode))
 
-    :rtype: ``str``
+        if size < 128:
+            raise ValueError('Unsafe key size: %s' % (size))
+
+        self.aes_key_string = aes_key_string
+        self.hmac_key_string = hmac_key_string
+        self.hmac_key_size = int(hmac_key_size)
+        self.mode = mode.upper()
+        self.size = int(size)
+
+        # We also store bytes version of the key since bytes are needed by encrypt and decrypt
+        # methods
+        self.hmac_key_bytes = Base64WSDecode(self.hmac_key_string)
+        self.aes_key_bytes = Base64WSDecode(self.aes_key_string)
+
+    @classmethod
+    def generate(self, key_size=256):
+        """
+        Generate a new AES key with the corresponding HMAC key.
+
+        :rtype: :class:`AESKey`
+        """
+        if key_size < 128:
```

Comment:
You may also want to consider putting the minimum AES key size into its own constant to give the reader an idea of where the number 128 came from. Right now it's just a magic number to anybody who doesn't keep up on crypto recommendations.

Reply:
Agreed, I will make a constant.